src/parser.ts

```diff
@@ -7,6 +7,14 @@ export type ParseParams = {
   seen: { schema: any; objects: any[] }[];
 };
 
+const ZodErrorHandler = (messageOrObject: string | object): void => {
+  if(typeof messageOrObject === 'string') {
+    throw ZodError.fromString(messageOrObject);
```

Comment:
maybe leave throw outside of the `zodErrorHandler` to not to break error stack trace?

Reply:
Good point to discuss from one side more clear stack trace from other side code duplication :(